kt/plugins/godot-gradle-plugin/src/main/kotlin/godot/gradle/tasks/initialJavaCompileForClassGraph.kt

```diff
@@ -0,0 +1,24 @@
+package godot.gradle.tasks
+
+import org.gradle.api.Project
+import org.gradle.api.tasks.SourceSetContainer
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.api.tasks.compile.JavaCompile
+
+fun Project.initialJavaCompileForClassGraph(): TaskProvider<JavaCompile> {
+    val mainSourceSet =  project.extensions
+        .getByType(SourceSetContainer::class.java)
+        .getByName("main")
+
+    val classGraphJavaCompile = tasks.register("intitialForClassGraphCompileJava", JavaCompile::class.java) {
+        it.source = mainSourceSet.allJava.asFileTree
```

Comment:
don't we limit the classgraph to the main source set like this? so adding separate source sets is not possible like this right?

Reply:
I think you're right. Do you know a better way to do ?

Reply:
Also already resolved in the separate branch. will resolve the conversation